diff --git a/system/core/run-as/run-as.cpp b/system/core/run-as/run-as.cpp
index b27cfad7c..c2672039a 100644
--- a/system/core/run-as/run-as.cpp
+++ b/system/core/run-as/run-as.cpp
@@ -190,6 +190,12 @@ int main(int argc, char* argv[]) {
     cmd_argv_offset += 2;
   }
 
+  bool force = false;
+  if ((argc >= cmd_argv_offset + 1) && !strcmp(argv[cmd_argv_offset], "--force")) {
+    force = true;
+    cmd_argv_offset += 1;
+  }
+
   // Retrieve package information from system, switching egid so we can read the file.
   gid_t old_egid = getegid();
   if (setegid(AID_PACKAGE_INFO) == -1) error(1, errno, "setegid(AID_PACKAGE_INFO) failed");
@@ -222,12 +228,12 @@ int main(int argc, char* argv[]) {
   uid_t userAppId = (AID_USER_OFFSET * userId) + info.uid;
 
   // Reject system packages.
-  if (userAppId < AID_APP) {
+  if (userAppId < AID_APP && !force) {
     error(1, 0, "package not an application: %s", pkgname);
   }
 
   // Reject any non-debuggable package.
-  if (!info.debuggable) {
+  if (!info.debuggable && !force) {
     error(1, 0, "package not debuggable: %s", pkgname);
   }
 
-- 
2.19.1

